autoload/ale/lsp/response.vim

```diff
@@ -27,6 +27,7 @@ function! ale#lsp#response#ReadDiagnostics(response) abort
 
     for l:diagnostic in a:response.params.diagnostics
         let l:severity = get(l:diagnostic, 'severity', 0)
+        let l:diagnostic.message = substitute(l:diagnostic.message, "\n", ' ', 'g')
```

Comment:
Instead of modifying the incoming Dictionary, set the value with newlines in the `detail` key, and the value with newlines replaced in the `text` key. Add tests for this.